core/src/main/java/org/mule/expression/ExpressionUtils.java

```diff
@@ -281,6 +282,48 @@ public static TypedValue getTypedProperty(String expression, MuleMessage msg, Pr
         return new TypedValue(result, propertyDataType);
     }
 
+    /**
+     * Returns {@code true} if the {@code expression} string starts with
```

Comment:
Javadocs should explain what the method does first and then explain what they return. Something like "checks if a value is an expression, ie: starts and ends with a expression prefix and sufix"